web/html/src/manager/minion/packages/package-states.js

```diff
@@ -1,399 +1,368 @@
+  // Use this one only for Select's:
+  // https://github.com/Semantic-Org/Semantic-UI-React/issues/638#issuecomment-252035750
+  const setUiView = () => {
+    return (event, data): void => {
+      setView(data);
     }
-    const request = Network.post(
-        "/rhn/manager/api/states/packages/save",
-        JSON.stringify({
-            sid: serverId,
-            packageStates: states
-        }),
-        "application/json"
-    ).promise.then(data => {
-      console.log("success: " + data);
-      const newPackageStates = data.map(state => {
-          state.packageStateId = normalizePackageState(state.packageStateId);
-          return state;
-      });
-
-      const newSearchResults = this.state.search.results.map( state => {
-        const changed = this.state.changed.get(packageStateKey(state))
-        if(changed !== undefined) {
-            return changed.value;
-        } else{
-            return state;
-        }
-      });
+  };
 
-      this.setState({
-        changed: new Map(),
-        view: "system",
-        search: {
-            filter: this.state.search.filter,
-            results: newSearchResults
-        },
-        packageStates: newPackageStates,
-        messages: MessagesUtils.info(t('Package states have been saved.'))
-      });
-    }, jqXHR => {
-      console.log("fail: " + jqXHR);
-      throw "failed to save";
-    });
-    return request;
-  }
-
-  applyPackageState() {
-    if (this.state.changed.size > 0) {
-        const response = confirm(t("There are unsaved changes. Do you want to proceed ?"))
-        if (response == false) {
-            return null;
-        }
+  const applyPackageState = () => {
+    if (changed.size > 0) {
+      const response = confirm(t("There are unsaved changes. Do you want to proceed ?"));
+      if (response === false) {
+        return null;
+      }
     }
 
-    const request = Network.post(
-        "/rhn/manager/api/states/apply",
-        JSON.stringify({
-            id: serverId,
-            type: "SERVER",
-            states: ["packages"]
-        }),
-        "application/json"
-    );
-    return request.promise.then(data => {
-          console.log("apply action queued:" + data);
-          this.setState({
-              messages: MessagesUtils.info(<span>{t("Applying the packages states has been ")}
-                  <a href={"/rhn/systems/details/history/Event.do?sid=" + serverId + "&aid=" + data}>{t("scheduled")}</a>
-              </span>)
-          });
+    fetchPackageStatesApi(action.APPLY, serverId)
+      .then(data => {
+        console.log("apply action queued:" + data);
+      }).catch(error => {
+      showErrorToastr(error, {autoHide: false});
     });
-  }
+  };
 
-  setView(view) {
-    return event => {
-      this.setState({
-          view: view
-      });
+  const save = (): Promise<any> => {
+    const toSave = [];
+    for (const state in changed) {
+      if (typeof changed[state].value === 'object') {
```

Comment:
why do we need this check?

Reply:
Because we would possibly access an object key without an object and this would result in an error. It is unlikely that this happens but now this possible error source is caught. (Also this was a Javascript warning in IntelliJ)